src/evo/providertx.h

```diff
@@ -21,16 +21,53 @@ class CBlockIndex;
 class CCoinsViewCache;
 class TxValidationState;
 
+class PayoutShare
+{
+public:
+    CScript scriptPayout{};
+    uint16_t payoutShareReward{0};
+    PayoutShare() = default;
+    explicit PayoutShare(const CScript& scriptPayout, const uint16_t& payoutShareReward = 10000) :
+        scriptPayout(scriptPayout), payoutShareReward(payoutShareReward){};
+    SERIALIZE_METHODS(PayoutShare, obj)
+    {
+        READWRITE(obj.scriptPayout,
+                  obj.payoutShareReward);
+    }
+    bool operator==(const PayoutShare& payoutShare) const
+    {
+        return (this->scriptPayout == payoutShare.scriptPayout && this->payoutShareReward == payoutShare.payoutShareReward);
+    }
+    bool operator!=(const PayoutShare& payoutShare) const
+    {
+        return !(*this == payoutShare);
+    }
+    [[nodiscard]] UniValue ToJson() const
+    {
+        UniValue ret(UniValue::VOBJ);
+        CTxDestination dest;
+        ret.pushKV("payoutAddress", ExtractDestination(scriptPayout, dest) ? EncodeDestination(dest) : "UNKNOWN");
+        ret.pushKV("payoutShareReward", payoutShareReward);
+        return ret;
+    }
+};
+
 class CProRegTx
 {
 public:
     static constexpr auto SPECIALTX_TYPE = TRANSACTION_PROVIDER_REGISTER;
     static constexpr uint16_t LEGACY_BLS_VERSION = 1;
     static constexpr uint16_t BASIC_BLS_VERSION = 2;
+    static constexpr uint16_t MULTI_PAYOUT_VERSION = 3;
 
-    [[nodiscard]] static constexpr auto GetVersion(const bool is_basic_scheme_active) -> uint16_t
+    [[nodiscard]] static constexpr auto GetVersion(const bool is_basic_scheme_active, const bool is_multi_payout_active) -> uint16_t
     {
-        return is_basic_scheme_active ? BASIC_BLS_VERSION : LEGACY_BLS_VERSION;
+        if (is_multi_payout_active) {
+            // multi payout is activated after basic scheme
```

Comment:
assert could cause app crash because. ptx.payouotShares is data that accepted from internet

```
ptx.nVersion = CProRegTx::GetVersion(!use_legacy, ptx.payoutShares.size() > 1);
```

Reply:
good point! 👍 pls ignore this suggestion then